src/linear_algebra/free_module/strong_rank_condition.lean

```diff
@@ -0,0 +1,54 @@
+/-
+Copyright (c) 2021 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca
+-/
+
+import linear_algebra.charpoly.basic
+import linear_algebra.invariant_basis_number
+
+/-!
+
+# Strong rank condition for commutative rings
+
+We prove that any nontrivial commutative ring satisfies `strong_rank_condition`.
+
+## Main result
+
+* `comm_ring_strong_rank_condition R` : `R` has the `strong_rank_condition`.
+
+## References
+
+We follow the proof given in https://mathoverflow.net/a/47846/7845.
```

Comment:
Could you sketch the proof very quickly here?

Reply:
Done.